pages/profile/[username].tsx

```diff
@@ -0,0 +1,70 @@
+import { TableContainer } from "@chakra-ui/table";
+import { DataTable } from "components/shared/DataTable";
+import { PageContainer } from "components/shared/PageContainer";
+import { Views } from "queries";
+import { getSupabaseClient } from "utils/getSupabaseClient";
+
+const dbClient = getSupabaseClient();
+
+export async function getStaticPaths() {
+  const { data } = await dbClient.from(Views.PublicSignups).select("username");
+
+  const payload = {
+    paths: data?.map(({ username }) => ({
+      params: { username },
+    })),
+    fallback: false, // can also be true or 'blocking'
+  };
+  return payload;
+}
+
+// `getStaticPaths` requires using `getStaticProps`
+export async function getStaticProps({
```

Comment:
For the getStaticProps, Why are there two Params? passed into the function parameters.  These params is it coming from the getStaticPath? And it is one of the many Usernames from the view table?

Reply:
There's only one param! the second thing you're seeing is the type declaration. The `username` is the only thing passed from `getStaticPaths` to `getStaticProps`